### ISSUE TITLE
Logging out from wpcom doesn't refresh the Sites screen properly

### ISSUE THREAD

**Issue by daniloercoli** [wordpress-mobile/WordPress-Android#3215]
My wpcom account has some .org Jetpack blogs linked with it.
When I disconnect from wpcom the sites screen doesn't get refreshed. The previously selected .com blog is already there. Even after logout.
See the gif below:
![last-blog](https://cloud.githubusercontent.com/assets/518232/10023515/d0b2a67c-6152-11e5-9a80-7b324d4be152.gif)

**Comment by daniloercoli** [comment]
I think issues related to login/logout from the app are all related to the fact we're synching Jetpack blogs with that call to `/me` endpoint during the initial signin.
Your Jetpack blogs are added to the app DB with the `dotcomFlag` set to 0. They are only used by Notifications IIRC. //cc @roundhill 

When you do the logout, the method [WordPressComSignOut](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/WordPress.java#L487) is called that in order calls [removeWpComUserRelatedData](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/WordPress.java#L563), that calls 
```
       // delete wpcom blogs
        wpDB.deleteWordPressComBlogs(context);
```
This last method, only deletes blogs that have the `dotcomFlag` set to 1, and so all previous Jetpack blogs are still there in the DB.

Later, [WordPressComSignOut](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/WordPress.java#L497) doesn't broadcast the `UserSignedOutCompletely` since it still sees those Jetpack blogs and the UI is not properly refreshed.
```
      // broadcast an event only if the user is completly signed out
        if (!AccountHelper.isSignedIn()) {
            EventBus.getDefault().post(new UserSignedOutCompletely());
        }
```

**Comment by roundhill** [comment]
> They are only used by Notifications IIRC. //cc @roundhill

That's correct, I'll see if there's an easy way to detect these and also delete them upon WordPress.com disconnect (without also deleting manually added self-hosted sites)

**Comment by roundhill** [comment]
Try out https://github.com/wordpress-mobile/WordPress-Android/commit/024b0dd8935f876b11ac070d238dff727ed84a32. I ran these tests:

 * Add wp.com user w/o jetpack sites and disconnect, should see sign in dialog.
 * Add wp.com user with jetpack sites and disconnect, should see sign in dialog.
 * Add wp.com user w/o jetpack sites AND sign in to a self-hosted site. When you disconnect you should only have the self-hosted site remaining.

**Comment by daniloercoli** [comment]
Thanks @roundhill! It worked fine.

**Comment by daniloercoli** [comment]
Addressed in #3237
